code/modules/mob/living/carbon/xenomorph/castes/wraith/abilities_wraith.dm

```diff
@@ -630,6 +631,18 @@ GLOBAL_LIST_INIT(wraith_banish_very_short_duration_list, typecacheof(list(
 		return
 	bullet.fire_at(shooter = bullet.firer, range = max(bullet.proj_max_range - bullet.distance_travelled, 0), angle = bullet.dir_angle, recursivity = TRUE, loc_override = get_turf(linked_portal))
 
+/// Handles slashing a portal to destroy it
+/obj/effect/wraith_portal/attacked_by(obj/item/I, mob/living/user, def_zone)
+	if(!I.sharp)
+		return
+	var/channel = SSsounds.random_available_channel()
+	playsound(user, "sound/effects/portal_opening.ogg", 30, channel = channel)
+	if(!do_after(user, 5 SECONDS, NONE, src))
+		user.stop_sound_channel(channel)
+		return
+	Destroy()
```

Comment:
```suggestion
	qdel(src)
```

Qdel is for deleting, destroy shouldnt be used since its not storing cleanup properly